src/Compilers/CSharp/Portable/Symbols/TypeSymbolWithAnnotations.cs

```diff
@@ -82,8 +148,10 @@ public static NullableAnnotation AsSpeakable(this NullableAnnotation annotation,
         /// </summary>
         public static NullableAnnotation JoinForFixingLowerBounds(this NullableAnnotation a, NullableAnnotation b)
         {
-            Debug.Assert(a.IsSpeakable());
```

Comment:
>Debug.Assert(a.IsSpeakable()); [](start = 12, length = 30)

If the assertions are removed here, can they be added to relevant inference (method, lambda returns, ternary) scenarios to guard that speakable types are inferred?

Reply:
I am working on a separate PR to totally eliminate the unspeakable NullableAnnotation values.  I'd like to defer the work until then.

---
In reply to: [260029279](https://github.com/dotnet/roslyn/pull/33648#discussion_r260029279) [](ancestors = 260029279)